### DISCUSSION TITLE
Retries in drake but not targets?

### DISCUSSION THREAD

**Discussion by JakeRuss** [ropensci/targets#517] [Help] [answered]
{drake}'s `make()` function has a retries argument, for re-building targets which result in errors, but it appears {targets} does not have the same feature. Have you decided this was a bad idea @wlandau ?

My simple use case is that I create pdf files via `tar_render()` in a pipeline and then I want to back them up onto an S3 bucket. I've written a small helper function that uploads files via `put_object` in {aws.s3} and I can use it in a `tar_target` step to upload the previously rendered pdf. 

But, in the event that upload fails, I'd have to re-run `tar_make()` to retry the upload. Do you recommend handling the upload outside of the {targets} pipeline?

**Comment by wlandau** [comment, marked as answer]
Retries were difficult to maintain in `drake` and did not add value to the majority of use cases. In `targets`, I recommend using something like https://github.com/randy3k/retry on a target-by-target basis: in your case, inside that helper function that calls `aws.s3::put_bucket()`. In other situations, you can use [hooks in `tarchetypes`](https://docs.ropensci.org/tarchetypes/reference/index.html#section-hooks) to apply https://github.com/randy3k/retry programmatically to multiple targets.

**Reply by JakeRuss** [reply]
Great, thanks for the explanation and suggestion.
